actionpack/test/dispatch/prefix_generation_test.rb

```diff
@@ -23,6 +23,24 @@ def to_model; self; end
     def persisted?; true; end
   end
 
+  class Author
+    extend ActiveModel::Naming
+
+    def to_param
+      "1"
+    end
+
+    def self.model_name
+      klass = +"Author"
+      def klass.name; self end
+
+      ActiveModel::Name.new(klass)
+    end
+
+    def to_model; self; end
+    def persisted?; true; end
+  end
```

Comment:
`Author` seems identical to `Post`.
I don't think it adds much to add a similar model.

Reply:
So it is definitely similar but the `Post` routes are explicitly defined here.  I set `author` to `resource` here to add it as a singular route.  I tried it with `Post` but the earlier `get` routes are seen first so it doesn't test properly and results in an error `Expected: "/awesome/blog/post"
  Actual: "/awesome/blog/posts/1"`:
```Ruby
    class BlogEngine < Rails::Engine
      routes.draw do
        get "/posts/:id", to: "inside_engine_generating#show", as: :post
        get "/posts", to: "inside_engine_generating#index", as: :posts
        get "/url_to_application", to: "inside_engine_generating#url_to_application"
        get "/polymorphic_path_for_engine", to: "inside_engine_generating#polymorphic_path_for_engine"
        get "/conflicting_url", to: "inside_engine_generating#conflicting"
        get "/foo", to: "never#invoked", as: :named_helper_that_should_be_invoked_only_in_respond_to_test

        get "/relative_path_root",       to: redirect("")
        get "/relative_path_redirect",   to: redirect("foo")
        get "/relative_option_root",     to: redirect(path: "")
        get "/relative_option_redirect", to: redirect(path: "foo")
        get "/relative_custom_root",     to: redirect { |params, request| "" }
        get "/relative_custom_redirect", to: redirect { |params, request| "foo" }

        get "/absolute_path_root",       to: redirect("/")
        get "/absolute_path_redirect",   to: redirect("/foo")
        get "/absolute_option_root",     to: redirect(path: "/")
        get "/absolute_option_redirect", to: redirect(path: "/foo")
        get "/absolute_custom_root",     to: redirect { |params, request| "/" }
        get "/absolute_custom_redirect", to: redirect { |params, request| "/foo" }

        resource :author
      end
    end
```

Reply:
Ah that makes sense.
Can we make that somehow more explicit?
Maybe renaming `Author` to `PolymorphicAuthor` or something `PostWithoutRoute`?
Or maybe a comment?

Reply:
I like comment! Renaming it to that.

Reply:
Haha no worries, sorry for misunderstanding